### PR TITLE
#VP-6684 revert breaking changes and add load dependecies

### PR DESCRIPTION
**Pull request by kostyrin**

### CODE REVIEW

File: src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs [hunk 1]
```diff
@@ -27,22 +27,19 @@ public MemberDocumentBuilder(IMemberService memberService, IDynamicPropertySearc
...
-            {
-                result.Add(await CreateDocument(member));
-            }
 
+            IList<IndexDocument> result = members.Select(CreateDocument).ToArray();
             return result;
         }
 
-        protected virtual Task<Member[]> GetMembers(IList<string> documentIds)
+        protected virtual async Task<Member[]> GetMembers(IList<string> documentIds)
```
Comment (asvishnyakov) [inline src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs:35]: You trying to fix one breaking change, but introduce another

File: src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs [hunk 2]
```diff
@@ -27,22 +27,19 @@ public MemberDocumentBuilder(IMemberService memberService, IDynamicPropertySearc
...
         {
-            return _memberService.GetByIdsAsync(documentIds.ToArray());
+            var result = await _memberService.GetByIdsAsync(documentIds.ToArray());
+            await LoadDependencies(result);
+            return result;
         }
 
-        protected async virtual Task<IndexDocument> CreateDocument(Member member)
+        protected virtual IndexDocument CreateDocument(Member member)
```
Comment (artem-dudarev) [inline src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs:42]: We already changed our module to override the async version.
You should not revert to sync again.

**Review by artem-dudarev — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs (hunk 2 above)
Comment (asvishnyakov) [inline src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs:42]: May we left create both CreateDocumentAsync and CreateDocument versions?

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs (hunk 2 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs:42]: @artem-dudarev What about others, who didn't upgraded already?

File: src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs (hunk 1 above)
Comment (kostyrin) [reply src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs:35]: Are you sure? please read the answer https://stackoverflow.com/a/44397025/4583735

File: src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs (hunk 2 above)
Comment (kostyrin) [reply src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs:42]: we can add 'Async' method

**Pull request closed without merging**